test/gtest/ucp/test_ucp_reconfigure.cc

```diff
@@ -0,0 +1,272 @@
+                                  test_owner),
+            m_cfg_index(UCP_WORKER_CFG_INDEX_NULL)
+        {
+        }
+
+        void connect(const ucp_test_base::entity *other,
+                     const ucp_ep_params_t &ep_params, int ep_idx = 0,
+                     int do_set_ep = 1) override;
+
+        void verify(const entity &other) const;
+
+        bool is_reconfigured() const
+        {
+            return m_cfg_index != ep()->cfg_index;
+        }
+
+        static const entity &to_reconfigured(const ucp_test_base::entity &e)
+        {
+            return *static_cast<const entity*>(&e);
+        }
+
+    private:
+        void store_config();
+        ucp_tl_bitmap_t ep_tl_bitmap() const;
+        address_pair_p get_address(bool ep_only) const;
+        bool has_matching_lane(ucp_ep_h ep, ucp_lane_index_t lane_idx,
+                               const entity &other) const;
+
+        ucp_worker_cfg_index_t m_cfg_index;
+        std::vector<uct_ep_h>  m_uct_eps;
+    };
+
+    void init() override
+    {
+        ucp_test::init();
+
+        /* Check presence of IB devices using rc_verbs, as all devices must
+         * support it. */
+        if (!has_resource(sender(), "rc_verbs")) {
+            UCS_TEST_SKIP_R("IB transport is not present");
+        }
+
+        m_entities.clear();
+    }
+
+public:
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        add_variant_values(variants, get_test_variants_feature, 0);
+        add_variant_values(variants, get_test_variants_feature, 1, "excl_if");
+    }
+
+    static void
+    get_test_variants_feature(std::vector<ucp_test_variant> &variants)
+    {
+        add_variant_with_value(variants, UCP_FEATURE_TAG, 0, "");
+    }
+
+    bool exclude_iface() const
+    {
+        return get_variant_value(1);
+    }
+
+    void create_entity()
+    {
+        m_entities.push_back(new entity(GetParam(), m_ucp_config,
+                                        get_worker_params(), this));
+    }
+
+    void send_recv()
+    {
+        static constexpr unsigned num_iterations = 100;
+        static constexpr size_t msg_size         = 16 * UCS_KBYTE;
+        const ucp_request_param_t param          = {
+            .op_attr_mask = UCP_OP_ATTR_FLAG_NO_IMM_CMPL
+        };
+        std::string sbuf(msg_size, 'a'), rbuf(msg_size, 'b');
+
+        sender().connect(&receiver(), get_ep_params());
+        receiver().connect(&sender(), get_ep_params());
+
+        for (unsigned i = 0; i < num_iterations; ++i) {
+            void *sreq = ucp_tag_send_nbx(sender().ep(), sbuf.c_str(), msg_size,
+                                          0, &param);
+            void *rreq = ucp_tag_recv_nbx(receiver().worker(),
+                                          (void*)rbuf.c_str(), msg_size, 0, 0,
+                                          &param);
+            request_wait(rreq);
+            request_wait(sreq);
+            EXPECT_EQ(sbuf, rbuf);

```

Comment:
After the first iteration this comparison is meaningless..

Reply:
what if the first iteration fails to fill the buffer and the others succeed?

Reply:
reuse `pattern_fill`/`pattern_check`